File: src/main/java/com/dh/ondot/member/core/exception/AlreadyOnboardedException.java
package com.dh.ondot.member.core.exception;

import com.dh.ondot.core.exception.ConflictException;

import static com.dh.ondot.core.exception.ErrorCode.ALREADY_ONBOARDED_MEMBER;

public class AlreadyOnboardedException extends ConflictException {
    public AlreadyOnboardedException(Long memberId) {
        super(ALREADY_ONBOARDED_MEMBER.getMessage().formatted(memberId));
    }

    @Override
    public String getErrorCode() {
        return ALREADY_ONBOARDED_MEMBER.name();
    }
}

File: src/main/kotlin/com/dh/ondot/member/core/exception/AlreadyOnboardedException.kt
package com.dh.ondot.member.core.exception

import com.dh.ondot.core.exception.ConflictException
import com.dh.ondot.core.exception.ErrorCode.ALREADY_ONBOARDED_MEMBER

class AlreadyOnboardedException(memberId: Long) :
    ConflictException(ALREADY_ONBOARDED_MEMBER.message.format(memberId)) {
    override val errorCode: String get() = ALREADY_ONBOARDED_MEMBER.name
}
